budgetportal/summaries.py

```diff
@@ -553,3 +557,39 @@ def get_aggregate_drilldowns(self):
             openspending_api.get_programme_name_ref(),
             openspending_api.get_econ_class_4_ref(),
         ]
+
+
+class BudgetedAndActualComparison(DepartmentBudgetData):
+    def __init__(self, department):
+        super().__init__(department)
+        self.financial_year = None
+
+    def get_dataset(self):
+        return self.department.get_budgeted_and_actual_comparison_dataset(
+            self.financial_year
+        )
+
+    def get_detail_aggregate_url(self):
+        openspending_api = self.get_openspending_api()
+        department_name = self.department.name
+        if openspending_api is not None:
+            department_ref = openspending_api.get_department_name_ref()
+            cuts = [
+                department_ref + ":" + "{}".format(department_name),
+            ]
+            aggregate_url = openspending_api.aggregate_url(cuts=cuts)
+
+            return aggregate_url
+        else:
+            return None
+
+    def get_detail_csv_urls(self):
+        urls = {}
+        for fy in FinancialYear.get_available_years():
+            self.financial_year = fy.slug
+            aggr_url = self.get_detail_aggregate_url()
+            if aggr_url is not None:
+                url = csv_url(aggr_url)
+                urls[fy.slug] = url
+
+        return urls
```

Comment:
you're supposed to have one of these classes to produce the query for the in-year data.

One instance per dataset (therefore one instance per year)

Reply:
done